google/cloud/forseti/services/inventory/base/cai_gcp_client.py

```diff
@@ -52,23 +92,101 @@ def session(self):
+
+        Returns:
+            dict: Billing Account IAM policy.
+        """
+        resource = self.dao.fetch_cai_asset(
+            ContentTypes.iam_policy,
+            'google.cloud.billing.BillingAccount',
+            '//cloudbilling.googleapis.com/{}'.format(account_id),
+            self.session)
+        if resource:
+            return resource
+        # Fall back to live API if the data isn't in the CAI cache.
+        return super(CaiApiClientImpl, self).fetch_billing_account_iam_policy(
+            account_id)
+
+    def iter_billing_accounts(self):
+        """Iterate visible Billing Accounts in an organization from GCP API.
 
         Yields:
-            dict: Generator of backend service.
+            dict: Generator of billing accounts.
         """
         resources = self.dao.iter_cai_assets(
             ContentTypes.resource,
-            'google.compute.BackendService',
+            'google.cloud.billing.BillingAccount',
+            '',  # Billing accounts have no parent resource.
+            self.session)
+        for account in resources:
+            yield account
+
+    def _iter_compute_resources(self, asset_type, project_number):
+        """Iterate Compute resources from Cloud Asset data.
+
+        Args:
+            asset_type (str): The Compute asset type to iterate.
+            project_number (str): id of the project to query.
+
+        Returns:
+            generator: A generator of resources from Cloud Asset data.
+        """
+        return self.dao.iter_cai_assets(
+            ContentTypes.resource,
+            'google.compute.{}'.format(asset_type),
             '//cloudresourcemanager.googleapis.com/projects/{}'.format(
                 project_number),
             self.session)
+
+    def iter_compute_autoscalers(self, project_number):
+        """Iterate Autoscalers from Cloud Asset data.
+
+        Args:
+            project_number (str): id of the project to query.
+
+        Yields:
+            dict: Generator of autoscaler resources.
+        """
+        resources = self._iter_compute_resources('Autoscaler', project_number)
+        for autoscaler in resources:
+            yield autoscaler
+
+    def iter_compute_backendbuckets(self, project_number):
+        """Iterate Backend buckets from Cloud Asset data.
+
+        Args:
+            project_number (str): id of the project to query.
+
+        Yields:
+            dict: Generator of backend bucket resources.
+        """
+        resources = self._iter_compute_resources('BackendBucket',
+                                                 project_number)
+        for backendbucket in resources:
+            yield backendbucket
+
+    def iter_compute_backendservices(self, project_number):
+        """Iterate Backend services from Cloud Asset data.
+
+        Args:
+            project_number (str): id of the project to query.
+
+        Yields:
+            dict: Generator of backend service.
+        """
+        cai_key_map = {
```

Comment:
Just by looking at this, it's hard to tell what the keys are mapping to.  How about clarifying by giving the variable a name such as, `cai_to_gcp_key_map`?  This it's really what we are mapping.